tests/test_mixture.py

```diff
@@ -705,3 +706,34 @@ def test_mixture_with_DiracDelta():
     logp_res = factorized_joint_logprob({M_rv: m_vv, I_rv: i_vv})
 
     assert m_vv in logp_res
+
+
+def test_switch_mixture():
+    srng = at.random.RandomStream(29833)
+
+    X_rv = srng.normal(-10.0, 0.1, name="X")
+    Y_rv = srng.normal(10.0, 0.1, name="Y")
+
+    I_rv = srng.bernoulli(0.5, name="I")
+    i_vv = I_rv.clone()
+    i_vv.name = "i"
+
+    Z1_rv = at.switch(I_rv, X_rv, Y_rv)
+    z1_vv = Z1_rv.clone()
+    z1_vv.name = "z1"
+
+    fgraph, _, _ = construct_ir_fgraph({Z1_rv: z1_vv, I_rv: i_vv})
+
+    assert isinstance(fgraph.outputs[0].owner.op, MixtureRV)
+
+    # building the identical graph but with a stack
+
+    Z2_rv = at.stack(X_rv, Y_rv)[I_rv]
+    z2_vv = Z2_rv.clone()
+    z2_vv.name = "z2"
+
+    z1_logp = joint_logprob({Z1_rv: z1_vv, I_rv: i_vv}).eval({z1_vv: -10, i_vv: 0})
+    z2_logp = joint_logprob({Z2_rv: z2_vv, I_rv: i_vv}).eval({z2_vv: -10, i_vv: 0})
+
+    np.testing.assert_almost_equal(0.69049938, z1_logp)
+    np.testing.assert_almost_equal(0.69049938, z2_logp)
```

Comment:
For these symbolic things we should always try to directly assert the kinds of graphs we're expecting to get back.  Functions like `aesara.graph.basic.equal_computations` are usually the most direct way to do that; however, in this case, since we're dealing with a single node that holds all the relevant information, we can just inspect it directly (e.g. assert that the values of `fgraph.outputs[0].owner.inputs` are the expected `Variable`s&mdash;or their clones).

Numeric evaluation might only be helpful to confirm that the results of the underlying `switch`-to-`stack` conversion are correct, but I'm not sure that's worth testing in Aesara (e.g. it can be done directly in NumPy).  
Aside from that, a `logprob` test isn't entirely necessary, since it's testing the same mixture functionality that's already being tested above&mdash;and if it isn't, it should be tested above.  Regardless, it's not bad to have.

Reply:
I believe that #158 is a blocker for this PR, more specifically in writing a test to assess the equivalency between mixture graphs defined via `switch` and `stack`.

Reply:
> I believe that #158 is a blocker for this PR, more specifically in writing a test to assess the equivalency between mixture graphs defined via `switch` and `stack`.

That shouldn't be a _blocker_; we can always check the graphs manually.